core/handlers/v2/hoverfly_postserveactiondetails_handler.go

```diff
@@ -0,0 +1,72 @@
+package v2
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/SpectoLabs/hoverfly/core/handlers"
+	"github.com/codegangsta/negroni"
+	"github.com/go-zoo/bone"
+)
+
+type HoverflyPostServeActionDetails interface {
+	GetPostServeActionDetails() PostServeActionDetailsView
+	RegisterPostServeActionHook(string, string, string, int) error
+	DeletePostServeActionHook(string) error
+}
+
+type HoverflyPostServeActionDetailsHandler struct {
+	Hoverfly HoverflyPostServeActionDetails
+}
+
+func (postServeActionDetailsHandler *HoverflyPostServeActionDetailsHandler) RegisterRoutes(mux *bone.Mux, am *handlers.AuthHandler) {
+
+	mux.Get("/api/v2/hoverfly/post-serve-actions", negroni.New(
+		negroni.HandlerFunc(am.RequireTokenAuthentication),
+		negroni.HandlerFunc(postServeActionDetailsHandler.Get),
+	))
+	mux.Post("/api/v2/hoverfly/post-serve-actions/hook", negroni.New(
```

Comment:
how about making this an idempotent endpoint and use `PUT` instead, it's how the middleware endpoint works, either create or update